packages/db/tests/translate/test-granules.js

```diff
@@ -1,7 +1,309 @@
+  fakeGranuleRecordFactory,
+  fakeExecutionRecordFactory,
+} = require('../../dist');
+
+const {
+  translateApiGranuleToPostgresGranule,
+  translatePostgresGranuleToApiGranule,
+} = require('../../dist/translate/granules');
+
+const { migrationDir } = require('../../../../lambdas/db-migration/dist/lambda');
+
+const testDbName = `granule_${cryptoRandomString({ length: 10 })}`;
+
+test.before(async (t) => {
+  const { knexAdmin, knex } = await generateLocalTestDb(
+    testDbName,
+    migrationDir
+  );
+  t.context.knexAdmin = knexAdmin;
+  t.context.knex = knex;
+
+  // Create collection
+  t.context.collectionPgModel = new CollectionPgModel();
+  const collection = fakeCollectionRecordFactory({ name: 'collectionName', version: 'collectionVersion' });
+  const [collectionCumulusId] = await t.context.collectionPgModel.create(knex, collection);
+
+  // Create provider
+  t.context.providerPgModel = new ProviderPgModel();
+  const provider = fakeProviderRecordFactory({ name: 'providerName' });
+  const [providerCumulusId] = await t.context.providerPgModel.create(knex, provider);
+
+  // Create PDR
+  t.context.pdrPgModel = new PdrPgModel();
+  const pdr = fakePdrRecordFactory({
+    name: 'pdrName',
+    collection_cumulus_id: collectionCumulusId,
+    provider_cumulus_id: providerCumulusId,
+  });
+  const [pdrCumulusId] = await t.context.pdrPgModel.create(knex, pdr);
+
+  // Create Granule
+  t.context.granulePgModel = new GranulePgModel();
+  [t.context.granuleCumulusId] = await t.context.granulePgModel.create(
+    knex,
+    fakeGranuleRecordFactory({
+      granule_id: cryptoRandomString({ length: 5 }),
+      status: 'running',
+      collection_cumulus_id: collectionCumulusId,
+      published: false,
+      duration: 10,
+      time_to_archive: 0,
+      time_to_process: 0,
+      product_volume: 1119742,
+      error: {},
+      cmr_link: cryptoRandomString({ length: 10 }),
+      pdr_cumulus_id: pdrCumulusId,
+      provider_cumulus_id: providerCumulusId,
+      beginning_date_time: new Date(Date.now() - 300 * 1000),
+      ending_date_time: new Date(Date.now() - 250 * 1000),
+      last_update_date_time: new Date(Date.now() - 100 * 1000),
+      processing_end_date_time: new Date(Date.now() - 500 * 1000),
+      processing_start_date_time: new Date(Date.now() - 400 * 1000),
+      production_date_time: new Date(Date.now() - 350 * 1000),
+      timestamp: new Date(Date.now() - 120 * 1000),
+      created_at: new Date(Date.now() - 200 * 1000),
+      updated_at: new Date(Date.now()),
+      query_fields: { foo: 'bar' },
+    })
+  );
+
+  // Create executions
+  const executionPgModel = new ExecutionPgModel();
+  const [executionACumulusId] = await executionPgModel.create(
+    knex,
+    fakeExecutionRecordFactory({ timestamp: new Date(Date.now()) })
+  );
+  const [executionBCumulusId] = await executionPgModel.create(
+    knex,
+    fakeExecutionRecordFactory({ timestamp: new Date(Date.now() - 200 * 1000) })
```

Comment:
Curious: was this explicitly set to match the created_at stamp?

Reply:
Oh no, that's just a coincidence. I'm trying to set all of the timestamp fields on each object to be different to ensure that I'm not mapping one field to another e.g. `processingEndDateTime: last_update_date_time`. Here it doesn't really matter so long as it's less than the other execution (the results are ordered). 

Updated with a different number:

https://github.com/nasa/cumulus/pull/2415/commits/d42c04e70380680f836f03cfe67df95369cc864a

We could also randomize these so it doesn't look like they're intentionally set to these numbers if you'd prefer.

Reply:
Not at all, just my pattern matching subroutines noted it and wondered if it was intentional for some reason I didn't grok.